66 pharmaverse story (#87)
pharmaverse/blog: posts/2023-08-30_pharmaverse_story/pharmaverse_story.qmd

---
title: "The pharmaverse (hi)story"
author:
  - name: Nicholas Eugenio
description: ""
date: "2023-08-30"
# please do not use any non-default categories.
# You can find the default categories in the repository README.md
categories: [community]
# feel free to change the image
image: "pharmaverse.png"
---

<!--------------- typical setup ----------------->

```{r setup, include=FALSE}
long_slug <- "2023-08-30_pharmaverse_story"
# renv::use(lockfile = "renv.lock")
```

<!--------------- post begins here ----------------->

Pharmaverse: from motivation to present

> **Human history and pharmaverse context**

Reviewer: I think for titles we should use the title markdown to keep it consistent.
Please use: 

`# Human history and pharmaverse context`
Or
`## Human history and pharmaverse context`

For most aspects of life, it doesn't make sense to think, act or work alone for two main reasons:

1.  You will spend more energy and time;

2.  Someone else may be facing (or have faced) the same situation.

The English poet John Donne used to say *"No man is an island entire of itself; every man is a piece of the continent, a part of the main;"*.
I can't disagree with him.
And I dare say that Ari Siggaard Knoph (Novo Nordisk), Michael Rimler (GSK), Michael Stackhouse (Atorus), Ross Farrugia (Roche), and Sumesh Kalappurakal (Janssen) can't disagree with him either.
They are the founders of [pharmaverse](https://pharmaverse.org/), members of its Council and kindly shared their memories of how independent companies, in mid-2020, worked together in the creation of a set of packages developed to support the clinical reporting pipeline.

If you are not familiar with this pipeline, the important thing to know is that, in a nutshell, pharmaceutical companies must follow a bunch of standardized procedures and formats (from Clinical Data Interchange Standards Consortium, [CDISC](https://www.cdisc.org/)) when submitting clinical results to Health Authorities.

> Focus on this: **different** companies seeking the **same** standards for outputs.

Reviewer: This sentence looks a bit like a note to yourself. 
Maybe write: 
The focus is on this: **different** companies ...

Paraphrasing Ross *Breaking boundaries through open-source collaboration* presentation in [R/Pharma 2022](https://www.youtube.com/watch?v=APMDOS4v9Hk) and thinking of the development of a new drug, we are talking about a "post-competitive" scenario: the drug has already been discovered and the companies should "just" produce and deliver standardized results.

**Clinical reporting outputs**

Rationally, we can say that companies face the same challenges in these steps of the process.

> Irrationally, we can also say they were working in silos on that before 2018.

Reviewer: I am not sure if I understand how saying this is irrational.

Just as Isaac Newton and Gottfried W. Leibniz [independently developed the theory of infinitesimal calculus](https://en.wikipedia.org/wiki/History_of_calculus), pharmaceutical companies were independently working on R solutions for this pipeline.

But on August 16 and 17 of the mentioned year above, they gathered at the first edition of [R/Pharma conference](https://rinpharma.com/) to discuss R and open-source tooling for drug development (the reasons why open-source is an advantageous approach can be found in this [post](https://pharmaverse.github.io/blog/posts/2023-07-20_cross_company_dev/cross_industry_dev.html) written by Stefan Thoma).
And according to Isabela Velásquez's article, [Pharmaverse: Packages for clinical reporting workflows](https://posit.co/blog/pharmaverse-packages-for-clinical-reporting-workflows/), one of the most popular questions in this conference was "Is the package code available or on [CRAN](https://cran.r-project.org/)?".

Well, many of them were.

> And not necessarily at that date, but just to mention a few: [*pharmaRTF*](https://atorus-research.github.io/pharmaRTF/index.html) and [*Tplyr*](https://atorus-research.github.io/Tplyr/index.html) from Atorus, [*r2rtf*](https://merck.github.io/r2rtf/) from Merck, [*rtables*](https://insightsengineering.github.io/rtables/main/) from Roche, and etc.

Reviewer: ```suggestion
And not necessarily at that date, but just to mention a few: [*pharmaRTF*](https://atorus-research.github.io/pharmaRTF/index.html) and [*Tplyr*](https://atorus-research.github.io/Tplyr/index.html) from Atorus, [*r2rtf*](https://merck.github.io/r2rtf/) from Merck, [*rtables*](https://insightsengineering.github.io/rtables/main/) from Roche, etc.
```

> The thing is that were almost 10000 other packages as well (today, almost 20000).

Reviewer: This sounds like there are 10 000 packages on CRAN related to clinical reporting.

Reviewer: Maybe you could clarify in this sentence that *overall* there were 10 000 packages available on CRAN.

And that took to another two questions:

1.  With this overwhelming number of packages on CRAN, how to find the ones related to solving "clinical reporting problems"?

2.  Once the packages were found, how to choose which one to use among those that have the same functional purpose?

So, again, companies re-started to working in silos to find those answers.
But now, in collaborative silos and with common goals: create extremely useful packages to solve pharmaceutical-specific gaps once and solve them well!

**First partnerships**

In 2020, Michael Stackhouse and Michael Rimler talked and formed a partnership between Atorus and GSK to develop a few more packages, including [*metacore*](https://github.com/atorus-research/metacore), to read, store and manipulate metadata for ADaMs/SDTMs in a standardized object; [*xportr*](https://github.com/atorus-research/xportr), to create submission compliant SAS transport files and perform pharma specific dataset level validation checks; and [*logrx*](https://github.com/pharmaverse/logrx) (ex-*timber*), to build log to support reproducibility and traceability of an R script.

> Around the same time, Thomas Neitmann (actual Denali Therapeutics, ex-Roche) and Rimler discovered that both were working with ADaM in R, so Ross and Rimler saw an opportunity there and GSK started their partnership with Roche to build and release [*admiral*](https://github.com/pharmaverse/admiral) package.

Reviewer: ```suggestion
Around the same time, Thomas Neitmann (currently at Denali Therapeutics, then at Roche) and Rimler discovered that both were working with ADaM in R, so Neitmann, Ross and Rimler saw an opportunity there and GSK started their partnership with Roche to build and release [*admiral*](https://github.com/pharmaverse/admiral) package.
```

The idea of working together, the sense of community, and the appetite from organizations built more and more, with incentive and priority established up into the programming heads council.

Janssen had a huge effort in building R capabilities going on as well, by releasing [*tidytlg*](https://github.com/pharmaverse/tidytlg) and [*envsetup*](https://github.com/pharmaverse/envsetup)), so eventually Rimler, Stackhouse and Ross formalized [pharmaverse](https://pharmaverse.org/) and formed the council, adding in Sumesh (Janssen) and Ari Siggaard Knoph (Novo Nordisk) later joined as the fifth council member.

**Release, growth and developments**  

> At the end of their presentation ["Closing the Gap: Creating an End to End R Package Toolkit for the Clinical Reporting Pipeline."](https://www.youtube.com/watch?v=NW9qp5QOlIU), in [R/Pharma 2021](https://rinpharma.com/event/rinpharma2021/), Ben Straub's (GSK) and Eli Miller's (Atorus) welcomed the community to the [pharmaverse](https://pharmaverse.org/), a new group of packages developed by various pharmaceutical companies to support open-source clinical workflows.

Reviewer: ```suggestion
At the end of their presentation ["Closing the Gap: Creating an End to End R Package Toolkit for the Clinical Reporting Pipeline."](https://www.youtube.com/watch?v=NW9qp5QOlIU), in [R/Pharma 2021](https://rinpharma.com/event/rinpharma2021/), Ben Straub (GSK) and Eli Miller (Atorus) welcomed the community to the [pharmaverse](https://pharmaverse.org/), a curated collection of packages developed by various pharmaceutical companies to support open-source clinical workflows.
```

2.  Once the packages were found, how to choose which one to use among those that have the same functional purpose?

So, again, companies re-started to working in silos to find those answers.
But now, in collaborative silos and with common goals: create extremely useful packages to solve pharmaceutical-specific gaps once and solve them well!

From Stackhouse's report, Michael Rimler and he talked and ultimately formed a partnership between GSK and Atorus to develop a few more packages, including [*metacore*](https://github.com/atorus-research/metacore), [*xportr*](https://github.com/atorus-research/xportr), and *timber* (which became [*logrx*](https://github.com/pharmaverse/logrx).

Around the same time, Thomas Neitmann (actual Denali Therapeutics, ex-Roche) and Rimler discovered that both were working with ADaM in R, so Ross and Rimler saw an opportunity there and GSK started their partnership with Roche to build and release [*admiral*](https://github.com/pharmaverse/admiral) package.

The idea of working together, the sense of community, and the appetite from organizations built more and more, with incentive and priority established up into the programming heads council.

> Janssen had a huge effort in building R capabilities going on as well, by releasing [*tidytlg*](https://github.com/pharmaverse/tidytlg) and [*envsetup*](https://github.com/pharmaverse/envsetup)), so eventually Rimler, Stackhouse and Ross formalized [pharmaverse](https://pharmaverse.org/) and formed the council, adding in Sumesh (Janssen).

Author: Added ;)

At the end of their presentation ["Closing the Gap: Creating an End to End R Package Toolkit for the Clinical Reporting Pipeline."](https://www.youtube.com/watch?v=NW9qp5QOlIU), in [R/Pharma 2021](https://rinpharma.com/event/rinpharma2021/), Ben Straub's (GSK) and Eli Miller's (Atorus) welcomed the community to the [pharmaverse](https://pharmaverse.org/), a new group of packages developed by various pharmaceutical companies to support open-source clinical workflows.

From the outset, the name [pharmaverse](https://pharmaverse.org/) was chosen so that it could be a neutral home, unrelated to any company.
Also, it was established as not being a consortium, which means that founders don't own, fund, or maintain the packages.
Some individuals and companies maintain them but don't take ownership of them.
The focus of [pharmaverse](https://pharmaverse.org/) early on, and today, is on inter organization cooperation, to build an environment where, if organizations identify that they have a joint problem that they want to solve, this is the right space to work on and release it.

> [Pharmaverse](https://pharmaverse.org/) has grown a lot, at the time of writing this post we have 75 packages hosted in [pharmaverse](https://pharmaverse.org/) [CRAN-like instance](https://pharmaverse.r-universe.dev/builds), and this has led to a partnership with [PHUSE](https://phuse.global/About_PHUSE) to get support from their organization and platform, and because they are eager to advance and support [pharmaverse](https://pharmaverse.org/) mission.

Reviewer 2: ```suggestion
[Pharmaverse](https://pharmaverse.org/) has grown a lot, at the time of writing this post we have >25 packages recommended in [pharmaverse](https://pharmaverse.org/e2eclinical/), and this has led to a partnership with [PHUSE](https://phuse.global/About_PHUSE) to get support from their organization and platform, and because they are eager to advance and support [pharmaverse mission](https://pharmaverse.org/charter/).
```
See https://github.com/pharmaverse/pharmaverse/issues/252 as FYI - i kept the exact number a bit looser as it changes all the time.
